core/templates/dev/head/pages/exploration_editor/UserEmailPreferencesService.js

```diff
@@ -0,0 +1,71 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview User exploration emails service for the exploration settings.
+ */
+
+oppia.factory('userExplorationEmailsHandler', [
+  '$http', '$q', 'explorationData', 'alertsService',
+  'UrlInterpolationService',
+  function(
+      $http, $q, explorationData, alertsService, UrlInterpolationService) {
+    var MESSAGE_TYPE_SUGGESTION = 'suggestion';
+    var MESSAGE_TYPE_FEEDBACK = 'feedback';
+    return {
+      init: function(
+          feedbackNotificationsMuted, suggestionNotificationsMuted) {
+        this.feedbackNotificationsMuted = feedbackNotificationsMuted;
+        this.suggestionNotificationsMuted = suggestionNotificationsMuted;
+      },
+      areFeedbackNotificationsMuted: function() {
+        return this.feedbackNotificationsMuted;
+      },
+      areSuggestionNotificationsMuted: function() {
+        return this.suggestionNotificationsMuted;
+      },
+      setFeedbackNotifications: function(mute) {
+        this.saveChangeToBackend({
+          message_type: MESSAGE_TYPE_FEEDBACK,
+          mute: mute
+        });
+      },
+      setSuggestionNotifications: function(mute) {
+        this.saveChangeToBackend({
+          message_type: MESSAGE_TYPE_SUGGESTION,
+          mute: mute
+        });
+      },
```

Comment:
I have used two methods here, so I don't need to inject the two constants in SettingsTab.js and the code is more readable there.